addons/project_stock/models/project_project.py

```diff
@@ -27,6 +27,11 @@ def _get_picking_action(self, action_name, picking_type=None):
             context['restricted_picking_type_code'] = picking_type
             if picking_type == 'outgoing':
                 context['default_partner_id'] = self.partner_id.id
+                if self.env.user.property_warehouse_id:
+                    context['default_picking_type_id'] = self.env.user.property_warehouse_id.out_type_id.id
+            if picking_type == 'incoming':
+                if self.env.user.property_warehouse_id:
+                    context['default_picking_type_id'] = self.env.user.property_warehouse_id.in_type_id.id
```

Comment:
We need to do this in the project_stock_account module because project_stock is not dependent on sale_stock.